lib/presentation/model/file/selected_presentation_file.dart

```diff
@@ -35,12 +35,17 @@
 // the Additional Terms applicable to LinShare software.
 
 import 'package:equatable/equatable.dart';
+import 'package:http_parser/http_parser.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/media_type_extension.dart';
 
 class SelectedPresentationFile extends Equatable {
   final String fileName;
   final int fileSize;
+  final MediaType _mediaType;
+  MediaType get mediaType => _mediaType ?? fileName.getMediaType();
 
-  SelectedPresentationFile(this.fileName, this.fileSize);
+  SelectedPresentationFile(this.fileName, this.fileSize, {MediaType mediaType})
```

Comment:
how about `SelectedPresentationFile(this.fileName, this.fileSize, {this._mediaType})`

Reply:
then we don't need `: _mediaType = mediaType;` part

Reply:
I've tried, but Dart doesn't allow us use optional with private property '_'